specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/unified_metrics_API.json

```
@@ -0,0 +1,859 @@
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MetricDefinitions_ListAtSubscriptionScope",
+        "description": "Lists the metric definitions for the subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/RegionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/MetricNamespaceParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric definitions",
+            "schema": {
+              "$ref": "#/definitions/SubscriptionScopeMetricDefinitionCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Subscription level Metric Definitions without filter": {
+            "$ref": "./examples/GetMultiResourceMetricDefinitions.json"
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MetricDefinitions_List",
+        "description": "Lists the metric definitions for the resource.",
+        "parameters": [
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/MetricNamespaceParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
```

```suggestion
              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
```

The "v1" part is causing a conflict, and the code won't generate. If they are the same type, this needs to be updated to say "v5". If it's a different ErrorResponse type, it needs to be named something other than ErrorResponse.

---

This should be resolved now with my latest changes @gracewilcox